File: functions/src/models/TakenLesson.ts
// src/models/TakenLesson.ts
import { Lesson } from "./Lesson";   // יבוא נכון

export interface TakenLesson {
  lessonId      : string
  ownerId       : string
  lesson        : Lesson               // עובד כי Lesson מיוצא מה‑model
  ownerName     : string
  ownerPhotoUrl : string
  canRate       : boolean
  takenAt       : FirebaseFirestore.FieldValue
}

File: app/src/main/java/com/example/exchangingprivatelessons/domain/model/TakenLesson.kt
package com.example.exchangingprivatelessons.domain.model

/**
 * Lesson that the **current** user קיבל אישור להצטרף אליו.
 *
 * ‑ ה‐ Lesson המלא (לצרכי UI)
 * ‑ פרטי המורה להצגה ברשימה
 * ‑ יכולת דירוג (business‑rule – נקבעה ע״י השרת)
 */
data class TakenLesson(
    val lesson        : Lesson,
    val ownerName     : String,
    val ownerPhotoUrl : String?,
    val takenAt       : Long,     // ✅ Long
    val canRate       : Boolean
)
